### PR TITLE
Initial project checkin

### PR DESCRIPTION
**Pull request by tabishop**
Initial delivery of open sourced projects from IBM that have been modified with correct copyright statements and other changes needed for life in the open source environment.

### CODE REVIEW

File: service-base-os/pom.xml
Comment (jillrdoty) [inline service-base-os/pom.xml:113]: I can't remember, did we do a review of all these dependencies to confirm they're Apache 2.0?

File: service-base-utils-bld-os/build.properties
Comment (jillrdoty) [inline service-base-utils-bld-os/build.properties]: Not sure what this file is for, but given the reference to RTC (?) and CSAR, might be worth getting rid of unless needed?

File: service-base-utils-bld-os/pom.xml
Comment (jillrdoty) [inline service-base-utils-bld-os/pom.xml]: Probably worth just cleaning up this commented line.

File: service-base-utils-bld-os/run_configuration/selectedProjectMavenCleanInstall.launch
Comment (jillrdoty) [inline service-base-utils-bld-os/run_configuration/selectedProjectMavenCleanInstall.launch:1]: This looks like an Eclipse thing, is it worth including?

File: service-base-utils-parent-os/whBldToolsInstall.bash
Comment (jillrdoty) [inline service-base-utils-parent-os/whBldToolsInstall.bash]: Not sure what this does or if this is a real person?

File: service-swagger-fragment-os/src/main/java/com/ibm/watson/solutions/api/listeners/SwaggerSetup.java
Comment (jillrdoty) [inline service-swagger-fragment-os/src/main/java/com/ibm/watson/solutions/api/listeners/SwaggerSetup.java:32]: I'm seeing mixed whitespace in these.  Would prefer to see it cleaned up - tangentially, I think we did something to align between mac and windows whitespace in our other project, should take a look to see if that's worthwhile here.

File: service-swagger-fragment-os/src/main/resources/META-INF/resources/documentation/api/oauth2-redirect.html
Comment (jillrdoty) [inline service-swagger-fragment-os/src/main/resources/META-INF/resources/documentation/api/oauth2-redirect.html:1]: Is everything under /src/main/resources/META-INF/resources generated here?  If so, would like to see it not committed.

File: service-base-utils-bld-os/pom.xml
Comment (tabishop) [reply service-base-utils-bld-os/pom.xml]: Agreed. I will clean this up. Good catch.

File: service-swagger-fragment-os/src/main/resources/META-INF/resources/documentation/api/oauth2-redirect.html
Comment (tabishop) [reply service-swagger-fragment-os/src/main/resources/META-INF/resources/documentation/api/oauth2-redirect.html:1]: This stuff is not generated and is how they included the OpenAPI UI code in the project. They way they do it here allows REST api projects to replace certain files like the index.html or service-Constants.js which is a function we will take advantage of withe the cohorting engine REST api.

File: service-swagger-fragment-os/src/main/java/com/ibm/watson/solutions/api/listeners/SwaggerSetup.java
Comment (tabishop) [reply service-swagger-fragment-os/src/main/java/com/ibm/watson/solutions/api/listeners/SwaggerSetup.java:32]: Agreed, will clean this up.

File: common-multipart-utils-os/.gitignore
Comment (dkwasny-ibm) [inline common-multipart-utils-os/.gitignore:35]: `**/.settings` should be ignored alongside the rest of the Eclipse files.

File: common-multipart-utils-os/.settings/.gitignore
Comment (dkwasny-ibm) [inline common-multipart-utils-os/.settings/.gitignore:1]: If we ignore `.settings`, then this file (and containing folder) should be deleted from the repo.

File: service-base-utils-bld-os/run_configuration/selectedProjectMavenCleanInstall.launch
Comment (tabishop) [reply service-base-utils-bld-os/run_configuration/selectedProjectMavenCleanInstall.launch:1]: These are handy utilities in eclipse, I would like to keep them and I think the common services team wants them as well.

File: service-base-utils-parent-os/whBldToolsInstall.bash
Comment (tabishop) [reply service-base-utils-parent-os/whBldToolsInstall.bash]: Confirmed that this file and the other whBldToolsXXX files can be removed. I will do that.

File: service-base-utils-bld-os/build.properties
Comment (tabishop) [reply service-base-utils-bld-os/build.properties]: Confirmed I can get rid of this file as well. Will remove.

File: common-multipart-utils-os/pom.xml
Comment (dkwasny-ibm) [inline common-multipart-utils-os/pom.xml]: Pedantic: Mixing space indentation into a file with tab indentation.
This may happen in other files as well.

File: service-base-os/src/main/java/com/ibm/watson/common/service/base/security/XssCharEscapes.java
Comment (dkwasny-ibm) [inline service-base-os/src/main/java/com/ibm/watson/common/service/base/security/XssCharEscapes.java:124]: Am I right in assuming all of these incomplete method stubs came from the original project?

File: service-swagger-fragment-os/src/main/java/com/ibm/watson/solutions/api/listeners/SwaggerSetup.java
Comment (dkwasny-ibm) [reply service-swagger-fragment-os/src/main/java/com/ibm/watson/solutions/api/listeners/SwaggerSetup.java:32]: A quick pass over the repo pointed me towards the following files as having DOS line endings.
```
$ find . -exec file {} \; | sort -u | grep CRLF
./service-base-utils-bld-os/pom.xml: exported SGML document text, ASCII text, with CRLF, LF line terminators
./service-base-utils-parent-os/pom.xml: exported SGML document text, ASCII text, with CRLF, LF line terminators
./service-encryption-os/pom.xml: exported SGML document text, ASCII text, with CRLF, LF line terminators
```

**Review by dkwasny-ibm — COMMENTED** [review]
I did a light pass over everything with a slight focus on the pom files.
GH appears to be somewhat hiding it, but I commented on a resolved discussion regarding line endings.

File: common-multipart-utils-os/.gitignore
Comment (tabishop) [reply common-multipart-utils-os/.gitignore:35]: I decided to go with the .gitignore convention that the original projects had because I wanted to keep things as consistent as possible so that it will make it easier for the common services team if/when they take over these projects as things will match their conventions.

File: common-multipart-utils-os/.settings/.gitignore
Comment (tabishop) [reply common-multipart-utils-os/.settings/.gitignore:1]: See above comments about keeping .gitignore convention

File: service-base-os/src/main/java/com/ibm/watson/common/service/base/security/XssCharEscapes.java
Comment (tabishop) [reply service-base-os/src/main/java/com/ibm/watson/common/service/base/security/XssCharEscapes.java:124]: Yes, came with original projects.

File: common-multipart-utils-os/pom.xml
Comment (tabishop) [reply common-multipart-utils-os/pom.xml]: Good catch, will clean up.

**Review by msargentibm — COMMENTED** [review]
Skimmed through files and sanity checked license/header information.

Other than Jill's outstanding question  about the dependency licenses, nothing stuck out to me.

**Review by jillrdoty — APPROVED** [review]

**Review by msargentibm — APPROVED** [review]

File: service-base-os/pom.xml
Comment (tabishop) [reply service-base-os/pom.xml:113]: We don't need to get approval for these beyond the WICKED approval to push to the git repo. We will eventually need to scan these and make sure that any other IBM products that want to use them can do so, but that will happen later once we have a build output we can scan.

**Pull request merged**
